### ISSUE TITLE
out of memory allocating memory

### ISSUE THREAD

**Issue by doehyunbaek** [titzer/wizard-engine#198]
Hi Ben,

When wizard runs this binary, `out of memory allocating memory` error is emitted:

```console
~ wizeng.x86-64-linux --version
Wizard Engine 24α.2278
~ wizeng.x86-64-linux /home/don/wasm-r3/benchmarks/wamr#2561/wamr#2561.wasm
/home/don/wasm-r3/benchmarks/wamr#2561/wamr#2561.wasm:0x0: out of memory allocating memory
```

The file is here:

https://github.com/doehyunbaek/wasm-r3/blob/rr-reduce/benchmarks/wamr%232561/wamr%232561.wasm

**Comment by titzer** [comment]
This module has a declared memory size of 65536 pages, i.e. 4GiB. Currently Wizard limits the memory to 30000 pages on non-x86-64 targets, but should allow 4GiB on x86-64 linux. I'll take a look into why.

**Comment by titzer** [comment]
This should be fixed by #205
